### ISSUE TITLE
Starting the VM fails (in Centos7 VM)

### ISSUE THREAD

**Issue by raptus93** [vmware-archive/pcfdev#87]
Hey!

Im on a CentOS7 vagrant box, that runs on my local win7, behind a company proxy and try to use PCF Dev. A proxy setting is set up, so I can access the internet from this point. I downloaded version v0.16 for linux from the pivotal site, installed the cf cli, VirtualBox and the pcfdev-plugin and ran
`./cf dev start`, navigating through the installation prompts.
In the end of all this, I got this:

> Downloading VM...
> Progress: |====================>| 100%
> VM downloaded
> Allocating 3072 MB out of 489 MB total system memory (358 MB free).
> Importing VM...
> Starting VM...
> FAILED
> Error: failed to start VM: ssh connection timed out: ssh: handshake failed: read tcp 127.0.0.1:41521->127.0.0.1:33129: read: connection reset by peer

How do I fix this? Another strange thing is, that `./cf dev status` tells me: 

> Running

If I try to login to cf with:
`./cf login -a api.local.pcfdev.io --skip-ssl-validation`
I get 

> API endpoint: api.local.pcfdev.io
> FAILED
> Server error, status code: 504, error code: 0, message:

I conclude, that 1.) the plugin did not start properly. 2.) The feedback of the ./cf dev status ist wrong (?).

How might I fix this problem? Does anyone else have this problem?

Thanks a lot!
Bob

**Comment by sclevine** [comment]
Hi @raptus93,

The incorrect reporting of the VM state in this situation is a known issue, and we have a story prioritized to fix it.

However, you seem to be running PCF Dev in a Vagrant VM that has very limited memory:

```
Allocating 3072 MB out of 489 MB total system memory (358 MB free).
```

PCF Dev requires at least 3 GB of free memory to run. The plugin should warn you when you don't have enough free memory. Was there a confirmation prompt before `Downloading VM...`?

Also, if your Vagrant VM is virtualized by Virtualbox, you may not have the requisite VT-x processor extensions to virtualize VMs inside of it.

To address both of these issues, I recommend running PCF Dev directly on your Windows 7 host.

Hope this helps,
Stephen

**Comment by raptus93** [comment]
Hey @sclevine,

wow, thanks a lot for the fast answer! Yes, there was the prompt, I didn't realise how serious the warning about the memory is, I thought it would be more like a recommendation. I'm sorry for that. In my context, it does not make sense to run it directly on Windows 7, it is some kind of experiment. I'm going to check the feasibility of your proposals, and report afterwards.

Thankful regards,
Bob

**Comment by sclevine** [comment]
It might be helpful to know that PCF Dev runs in a very isolated manner on the host. The CLI plugin does roughly what Vagrant does -- spin up a VM with the correct Virtualbox networking interfaces.

**Comment by raptus93** [comment]
Hey again!
The first issue is now no problem anymore, I just assigned more memory to the CentOS7 VM and the problem was fixed. But I still cannot start PCF Dev. I read, that nested VT-x is not supported by VirtualBox right now, even though there is a feature request for [this](https://www.virtualbox.org/ticket/4032) (fun fact: the oldest post is from 2009). Anyway, the error is still this one 

> Error: failed to start VM: ssh connection timed out: ssh: handshake failed: read tcp 127.0.0.1:41085->127.0.0.1:34740: read: connection reset by peer

I will have to think about using VMware instead of VirtualBox for this issue.
Thanks for the help.
